File: src/my_cpp_pkg/src/add_two_ints_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using std::placeholders::_1;
using std::placeholders::_2;

using std::bind;

using example_interfaces::srv::AddTwoInts;

class AddTwoIntsServerNode : public rclcpp::Node
{
public:
    AddTwoIntsServerNode() : Node("add_two_ints_server")
    {
        server_ = this->create_service<AddTwoInts>("add_two_ints", bind(&AddTwoIntsServerNode::callbackTwoInts, this, _1, _2));
        RCLCPP_INFO(this->get_logger(), "Add Two Ints server has been started");
    }

private:
    void callbackTwoInts(const AddTwoInts::Request::SharedPtr request,const AddTwoInts::Response::SharedPtr response)
    {
        response->sum = request->a + request->b;
        RCLCPP_INFO(this->get_logger(), "%d + %d = %d", request->a, request->b, response->sum);
    }

    rclcpp::Service<AddTwoInts>::SharedPtr server_;

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    //shared pointer, you don't need to delete manually the instance when object is not referenced (like GC in Java)
    auto node = std::make_shared<AddTwoIntsServerNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/add_two_ints_server.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts

class AddTwoIntsServerNode(Node):

    def __init__(self):
        super().__init__("add_two_ints_server") # Node is NOT the executable, therefore name of the node doesn't have to have same name of the file 
        self.server_ = self.create_service(AddTwoInts, "add_two_ints", self.callback_add_two_ints)
        self.get_logger().info("Add two ints server has been started")
    
    def callback_add_two_ints(self, request, response):
        response.sum = request.a + request.b
        self.get_logger().info("{0} + {1} = {2}".format(request.a, request.b, response.sum))
        return response # if you forget to return response, you'll get an error when calling service

def main(args=None):
    rclpy.init(args=args) # first line to be written in any ROS2 .py node
    node = AddTwoIntsServerNode()
    rclpy.spin(node) # allow node to continue to be alive
    rclpy.shutdown() # last line of any ROS2 .py node

if __name__ == "__main__":
    main()